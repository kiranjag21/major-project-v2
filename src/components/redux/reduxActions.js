export const addItems = (selectedDish) => ({
  type: 'ADD_ITEMS',
  payload: selectedDish
});

export const changeQuantity = (id,quantity)=>({
type: 'CHANGE_QUANTITY',
payload : {
  id:id,
  quantity:quantity
}
});

export const removeItem = (id)=>({
type:'REMOVE_ITEM',
payload:id
});

export const clearCart =()=>({
type:'CLEAR_CART',

})
export const addUser = (id) => ({
  type: 'ADD_USER',
  payload: id
});
export const removeUser = () => ({
  type: 'REMOVE_USER'
});
export const setConnection = (con) => ({

  type: 'ADD_CONNECTION',
  payload: con
});

export const disconnect = () => ({

  type: 'DISCONNECT'
});