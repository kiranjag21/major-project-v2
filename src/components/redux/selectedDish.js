
export const SelectedDish = (state = {
  selectedDish: []
}, action) => {
// let selectedDish = state.selectedDish;
  switch (action.type) {

      case 'ADD_ITEMS':
          return { ...state, selectedDish: action.payload };

      case 'CHANGE_QUANTITY':
         console.log(state.selectedDish.dishes);
        let item = state.selectedDish.dishes.find((item) => item.dishId === action.payload.id);
        let newCart = state.selectedDish.dishes.filter((item) => item.dishId !== action.payload.id);
        item.quantity = action.payload.quantity;

        newCart.push(item);
        state.selectedDish.dishes=newCart;
        // console.log(newCart);
        return{...state,selectedDish:state.selectedDish};

         case 'REMOVE_ITEM':
           console.log("aaya");
           console.log(state.selectedDish.dishes);
           console.log(action.payload.id);
           let newCart1 = state.selectedDish.dishes.filter((item) => item.dishId !== action.payload);
            console.log(newCart1);
            state.selectedDish.dishes=newCart1;
            console.log(newCart1);
            return{...state,selectedDish:state.selectedDish};

         case 'CLEAR_CART':
          let selectedDish=[];
         return{...state,selectedDish:selectedDish};

      default:
          return state;
  }
}