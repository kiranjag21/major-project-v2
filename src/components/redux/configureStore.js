
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { Connection } from './pusherConnection';
import { SelectedDish } from './selectedDish';
import { Auth } from './auth';
export const configureStore = () => {
    const store = createStore(
        combineReducers({
            selectedDish: SelectedDish,
            connection: Connection,
            auth: Auth
        }),
        applyMiddleware(thunk, logger)
    );

    return store;
}