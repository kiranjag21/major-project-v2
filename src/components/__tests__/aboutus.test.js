import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";
import { Provider } from 'react-redux';
import { configureStore } from '../redux/configureStore';
import AboutUs from './../About Us/aboutus';

const store=configureStore();
test('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Provider store={store}><AboutUs></AboutUs></Provider></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
