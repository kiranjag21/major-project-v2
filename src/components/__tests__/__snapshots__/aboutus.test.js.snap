// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "fontFamily": "Poppins, sans-serif",
    }
  }
>
  <div
    classNames="bg py-3"
  >
    <div
      className="container py-3"
    >
      <div
        className="row mb-4"
      >
        <div
          className="col-lg-2"
        >
          <h2
            className="display-4 font-weight-light"
            style={
              Object {
                "color": "#1f2833",
              }
            }
          >
            GUIDES
          </h2>
          <span
            className="font-italic text-muted "
          >
            The torches to our Major Project Journey.
          </span>
        </div>
      </div>
      <div
        className="row text-center"
      >
        <div
          className="col-xl-3 col-sm-6 mb-5"
        >
          <div
            className="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              className="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-mentor.png"
              width="100"
            />
            <h5
              className="mb-0"
            >
              Saju Sasidharan
            </h5>
            <span
              className="small text-uppercase text-muted"
            >
              Mentor
            </span>
            <ul
              className="social mb-0 list-inline mt-3"
            >
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-xl-3 col-sm-6 mb-5"
        >
          <div
            className="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              className="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-trainer.png"
              width="100"
            />
            <h5
              className="mb-0"
            >
              Asfiya Khan
            </h5>
            <span
              className="small text-uppercase text-muted"
            >
              Trainer
            </span>
            <ul
              className="social mb-0 list-inline mt-3"
            >
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                className="list-inline-item"
              >
                <a
                  className="social-link"
                  href="#"
                >
                  <em
                    className="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container py-3"
    >
      <div
        class="row mb-4"
      >
        <div
          class="col-lg-2"
        >
          <h2
            class="display-4 font-weight-light"
            style={
              Object {
                "color": "#1f2833",
              }
            }
          >
            TEAM
          </h2>
          <span
            class="font-italic text-muted"
          >
            Together we Stand.
          </span>
        </div>
      </div>
      <div
        class="row text-center"
      >
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-8.jpg"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Mrunal
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-5.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Kiran
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-2.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Suraj
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-1.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Tanmay
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-6.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Pranjal
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-7.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Qusai
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin "
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-xl-3 col-sm-6 mb-5"
        >
          <div
            class="bg-white rounded shadow-sm py-5 px-4 peopleCarouselImg "
          >
            <img
              alt=""
              class="img-fluid rounded-circle mb-3 img-thumbnail shadow-sm"
              src="assets/img/avatar-5.png"
              width="100"
            />
            <h5
              class="mb-0"
            >
              Varun
            </h5>
            <span
              class="small text-uppercase text-muted"
            />
            <ul
              class="social mb-0 list-inline mt-3"
            >
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-facebook-f"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-twitter"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-instagram"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
              <li
                class="list-inline-item"
              >
                <a
                  class="social-link"
                  href="#"
                >
                  <em
                    class="fa fa-linkedin"
                    style={
                      Object {
                        "color": "#1f2833",
                        "fontSize": "15px",
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
